wgpu-core/src/snatch.rs

```diff
@@ -76,7 +85,24 @@ impl SnatchLock {
     }
 
     /// Request read access to snatchable resources.
+    #[track_caller]
     pub fn read(&self) -> SnatchGuard {
+        if cfg!(debug_assertions) {
+            let caller = Location::caller();
+            let backtrace = Backtrace::force_capture();
```

Comment:
capturing a backtrace can be very expensive operation and we do a lot of snatchlock reading accross the codebase. People should expect debug assertions to not degrate performance that much, so let's do only
```suggestion
            let backtrace = Backtrace::capture();
```
to make the backtrace opt-in like most backtraces in Rust.